### PR TITLE
Feat/assistant enhancements

### LINKED ISSUE #341: enhance assistant page
This issue aims to enhance the AI Assistant Screen in order to be more User Friendly.

### LINKED ISSUE #342: Bug: Assistant screen displays already added activities when go back button is pressed
To reproduce:
- on the assistant screen, press 'go'
- click on the first activity expand icon, then 'Add' button
- click the go back button

--> the UI displays for a short amount of time the activities that were added to the trip and should not have been displayed anymore'
https://github.com/user-attachments/assets/34b32874-d2fc-4591-82e0-1b307aced4a8

### PR DESCRIPTION
**Pull request by ioanajianu**
## Summary

This change improves the UI/UX of the Assistant Screen. The improvements include:
- better UI for the Ask Assistant button on the Schedule screen and handling at this stage of no internet connection
- better UI for the prompt TexField. It now shows 3 lines of text instead of 1, and pressing enter still sends the request.
- a text is shown when the page is first opened with basic instructions on how to use the assistant
- in case of an error, a generic error message is displayed instead of the whole error message. The details appear in the logs.
- the prompt sent to the API now contains the trip type as well, and the hardcoded strings from the `generatePrompt` functions where moved to strings res

## Changes

- **Screens/UI:** Changes live on the Schedule screen and Assistant screen, see above
- **Bug Fixes/Improvements:** This change also resolves #342 
- **Tests:** `AssistantUtils` unit tests have been changed to android tests because of the context that's needed to access the string resources

## Checklist

- [x] This PR resolves #341 
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

## Screenshots (if applicable)

https://github.com/user-attachments/assets/7a768703-5a8a-4188-97ae-224a5ff3cb38

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt [hunk 1]
```diff
@@ -264,14 +335,17 @@ fun SettingsDialog(
...
             Text(stringResource(R.string.provide_final_activities), modifier = Modifier.weight(1f))
             Switch(
                 checked = provideDraftActivities,
                 onCheckedChange = onProvideFinalActivitiesChanged,
                 modifier = Modifier.testTag("provideFinalActivitiesSwitch"))
           }
           Spacer(modifier = Modifier.height(4.dp))
+
           Row(verticalAlignment = Alignment.CenterVertically) {
+            // Switch to use only the current user's interests
```
Comment (mirceagavrizi) [inline app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt:348]: Here in the dialog the text is very close to the button ("interests" is almost touching the button). Maybe we can change it to "your interests" / "your user's interests" so it's a bit shorter.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -42,13 +42,23 @@
...
     <string name="prompt">Prompt</string>
-    <string name="go">Go</string>
+    <string name="ask">Ask</string>
     <string name="settings">Settings</string>
     <string name="settings_subtitle">Remember to click on the \'go\' button after changing the settings</string>
     <string name="provide_final_activities">Provide final activities with date and time (recommended for trips shorter than a week)</string>
     <string name="use_user_interests">Use only your profile\'s interests</string>
     <string name="use_participants_interests">Use all participants\'s interests</string>
     <string name="close">Close</string>
+    <string name="assistant_initial_screen">Please enter a prompt to receive activity recommendations. You can mention what types of activities you would like to do on your trip, or leave it empty for general recommendations based on the trip details. Fee free to use the settings button to customize the recommendations.</string>
```
Comment (mirceagavrizi) [inline app/src/main/res/values/strings.xml:52]: Here there is a small typo in the last sentence: Fee instead of Feel

Comment (mirceagavrizi) [inline app/src/main/res/values/strings.xml:52]: Here there is a small typo in the last sentence: Fee instead of Feel

Comment (mirceagavrizi) [inline app/src/main/res/values/strings.xml:52]: Also in the string "Remember to click ..." the button is now ask not go

**Review by mirceagavrizi — COMMENTED** [review]
I left some comments, just some nits and typos, otherwise great work on adding documentation and keeping the code coverage, as well as massively improving the UX for the assistant, it is way easier and more friendly. Let me know when you make the modifications and we'll be able to merge this PR.

**Review by mirceagavrizi — APPROVED** [review]
Thank you for taking into account my suggestions and quickly fixing them. Great job!

File: app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt (hunk 1 above)
Comment (ioanajianu) [reply app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt]: Thanks for pointing this out! I added a spacer between the text and the switch

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (ioanajianu) [reply app/src/main/res/values/strings.xml]: fixed!

Comment (ioanajianu) [reply app/src/main/res/values/strings.xml]: Done!

**Pull request merged**